Modules/ITS/src/ITSThresholdCalibrationTask.cxx

```diff
@@ -0,0 +1,456 @@
+      string data = split(info, ":")[1];
+      if (name == "Hs_pos") {
+        result.Hs = std::stod(data);
+      } else if (name == "Hic_Pos") {
+        result.HIC = std::stod(data);
+      } else if (name == "ChipID") {
+        result.ChipID = std::stod(data);
+      } else if (name == "VCASN") {
+        result.VCASN = std::stof(data);
+      } else if (name == "Rms") {
+        result.RMS = std::stof(data);
+      } else if (name == "Status") {
+        result.status = std::stod(data);
+      } else if (name == "Noise") {
+        result.Noise = std::stof(data);
+      } else if (name == "NoiseRMS") {
+        result.NoiseRMS = std::stof(data);
+      } else if (name == "ITHR") {
+        result.ITHR = std::stof(data);
+      } else if (name == "THR") {
+        result.THR = std::stof(data);
+      }
+    }
+  }
+  return result;
+}
+
+void ITSThresholdCalibrationTask::endOfCycle()
+{
+
+  std::ifstream runNumberFile(mRunNumberPath.c_str());
+  if (runNumberFile) {
+
+    std::string runNumber;
+    runNumberFile >> runNumber;
+    if (runNumber != mRunNumber) {
+      for (unsigned int iObj = 0; iObj < mPublishedObjects.size(); iObj++)
+        getObjectsManager()->addMetadata(mPublishedObjects.at(iObj)->GetName(), "Run", runNumber);
+      mRunNumber = runNumber;
+    }
+    ILOG(Info, Support) << "endOfCycle" << ENDM;
+  }
+}
+
+void ITSThresholdCalibrationTask::endOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "endOfActivity" << ENDM;
+}
+
+void ITSThresholdCalibrationTask::reset()
+{
+  ILOG(Info, Support) << "Resetting the histogram" << ENDM;
+  hSuccessRate->Reset();
+
+  for (int iScan = 0; iScan < 3; iScan++) {
+    for (int iLayer = 0; iLayer < 7; iLayer++) {
+      hCalibrationLayer[iLayer][iScan]->Reset();
+      hCalibrationRMSLayer[iLayer][iScan]->Reset();
+
+      if (iScan == 2) {
+        hCalibrationThrNoiseLayer[iLayer]->Reset();
+        hCalibrationThrNoiseRMSLayer[iLayer]->Reset();
+      }
+    }
+    for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+      hCalibrationChipSum[iScan][iBarrel]->Reset();
+      hCalibrationChipAverage[iScan][iBarrel]->Reset();
+      hCalibrationChipCounts[iScan][iBarrel]->Reset();
+
+      hCalibrationRMSChipSum[iScan][iBarrel]->Reset();
+      hCalibrationRMSChipAverage[iScan][iBarrel]->Reset();
+    }
+  }
+
+  for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+
+    hCalibrationThrNoiseChipSum[iBarrel]->Reset();
+    hCalibrationThrNoiseChipAverage[iBarrel]->Reset();
+
+    hCalibrationThrNoiseRMSChipSum[iBarrel]->Reset();
+    hCalibrationThrNoiseRMSChipAverage[iBarrel]->Reset();
+  }
+}
+
+void ITSThresholdCalibrationTask::createAllHistos()
+{
+  /*
+    TString sScanTypes[3]= {"VCASN","ITHR","THR"};
+    TString sBarrelType[3]= {"IB","MB","OB"};
+    Int_t nChips[3]= {9,112, 196};
+    Int_t nStaves[3]={48,54,90};
+    Int_t nXmax[3]= {80,100,400};
+    TString sXtitles[3]={"DAC","DAC","e"};
+  */
+  for (int iScan = 0; iScan < 3; iScan++) {
```

Comment:
General comment for all plots: give "shifter friendly" titles to plot. For example: "Average chip thresholds for IB" or "Average chip ITHR for MB", ..... The histo names are fine.